extensions/notebook/src/common/notebookUtils.ts

```diff
@@ -4,164 +4,154 @@
- * Defaults to 24 bytes, which creates a 48-char hex string
- */
-export function getRandomToken(size: number = 24): Promise<string> {
-	return new Promise((resolve, reject) => {
-		crypto.randomBytes(size, (err, buffer) => {
-			if (err) {
-				reject(err);
-			}
-			let token = buffer.toString('hex');
-			resolve(token);
-		});
-	});
-}
+export class NotebookUtils {
 
-export async function newNotebook(connectionProfile?: azdata.IConnectionProfile): Promise<azdata.nb.NotebookEditor> {
-	const title = findNextUntitledEditorName();
-	const untitledUri = vscode.Uri.parse(`untitled:${title}`);
-	const options: azdata.nb.NotebookShowOptions = connectionProfile ? {
-		viewColumn: null,
-		preserveFocus: true,
-		preview: null,
-		providerId: null,
-		connectionProfile: connectionProfile,
-		defaultKernel: null
-	} : null;
-	return azdata.nb.showNotebookDocument(untitledUri, options);
-}
+	constructor(private _apiWrapper: ApiWrapper) { }
 
-function findNextUntitledEditorName(): string {
-	let nextVal = 0;
-	// Note: this will go forever if it's coded wrong, or you have infinite Untitled notebooks!
-	while (true) {
-		let title = `Notebook-${nextVal}`;
-		if (isEditorTitleFree(title)) {
-			return title;
+	public async newNotebook(connectionProfile?: azdata.IConnectionProfile): Promise<azdata.nb.NotebookEditor> {
+		const title = this.findNextUntitledEditorName();
+		const untitledUri = vscode.Uri.parse(`untitled:${title}`);
+		const options: azdata.nb.NotebookShowOptions = connectionProfile ? {
+			viewColumn: null,
+			preserveFocus: true,
+			preview: null,
+			providerId: null,
+			connectionProfile: connectionProfile,
+			defaultKernel: null
+		} : null;
+		return azdata.nb.showNotebookDocument(untitledUri, options);
+	}
+
+	private findNextUntitledEditorName(): string {
+		let nextVal = 0;
+		// Note: this will go forever if it's coded wrong, or you have infinite Untitled notebooks!
+		while (true) {
+			let title = `Notebook-${nextVal}`;
+			if (isEditorTitleFree(title)) {
+				return title;
+			}
+			nextVal++;
 		}
-		nextVal++;
 	}
-}
 
-export async function openNotebook(): Promise<void> {
-	try {
-		let filter: { [key: string]: Array<string> } = {};
-		// TODO support querying valid notebook file types
-		filter[localize('notebookFiles', "Notebooks")] = ['ipynb'];
-		let file = await vscode.window.showOpenDialog({
-			filters: filter
-		});
-		if (file) {
-			let doc = await vscode.workspace.openTextDocument(file[0]);
-			vscode.window.showTextDocument(doc);
+	public async openNotebook(): Promise<void> {
+		try {
+			let filter: { [key: string]: Array<string> } = {};
+			// TODO support querying valid notebook file types
+			filter[localize('notebookFiles', "Notebooks")] = ['ipynb'];
+			let file = await this._apiWrapper.showOpenDialog({
+				filters: filter
+			});
+			if (file && file.length > 0) {
+				await azdata.nb.showNotebookDocument(file[0]);
```

Comment:
This is the fix. Instead of calling openTextDocument (which was known to be broken in the last vs code merge) just call the API we have already.